Plugins/Sycl/include/Acts/Plugins/Sycl/Seeding/Seedfinder.hpp

```diff
@@ -0,0 +1,81 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+// Acts include(s).
+#include "Acts/Plugins/Sycl/Seeding/DeviceExperimentCuts.hpp"
+#include "Acts/Plugins/Sycl/Seeding/detail/Types.h"
+#include "Acts/Seeding/InternalSeed.hpp"
+#include "Acts/Seeding/InternalSpacePoint.hpp"
+#include "Acts/Seeding/SeedFilter.hpp"
+#include "Acts/Seeding/SeedfinderConfig.hpp"
+
+/// Forward declaration of incomplete type cl::sycl::queue
+inline namespace cl {
+namespace sycl {
+class queue;
+}
+};  // namespace cl
+
+namespace Acts::Sycl {
+
+/// @brief Seedfinding algorithm implemented with SYCL to offload computations
+/// to GPUs.
+void createSeedsForGroupSycl(
+    const std::shared_ptr<cl::sycl::queue>& q,
+    const detail::deviceSeedfinderConfig& configData,
+    const DeviceExperimentCuts& deviceCuts,
+    const std::vector<detail::deviceSpacePoint>& bottomSPs,
+    const std::vector<detail::deviceSpacePoint>& middleSPs,
+    const std::vector<detail::deviceSpacePoint>& topSPs,
+    std::vector<std::vector<detail::SeedData>>& seeds);
+
+/// @brief This function creates the SYCL queue object.
+///
+/// SYCL implementation details are hidden from this class, that's why we're
+/// creating the queue in a standalone translation unit.
+///
+/// @return A pointer to the queue.
+std::shared_ptr<cl::sycl::queue> createQueue(const std::string&);
```

Comment:
As we discussed before, I'd prefer to have the definitions of both of these functions in their own header files.

Reply:
Oh yes, I remember now.